### ISSUE TITLE
Support for aws-cn and aws-us-gov

### ISSUE THREAD

**Issue by gyaneshwar** [StateFarmIns/terraform-aws-default-log-retention#10]
Note: The verbiage expressed here are that of the author and are not representative of State Farm.

**Is your feature request related to a problem? Please describe.**
Applying for AWS china, this gives an issue, since china has `aws-cn` instead of `aws` as resource reference.
also for aws us gov regulation consider a different resource name as in below link.
https://docs.aws.amazon.com/govcloud-us/latest/UserGuide/using-govcloud-arns.html

```
Error: 1 error occurred: * creating inline policy (log_retention): MalformedPolicyDocument: Partition "aws" is not valid for resource "arn:aws:logs:*:*:*". status code: 400, request id: 000-0000-00000-0000000
with module.working.module.default_log_retention.aws_iam_role.log_retention
on .terraform/modules/working.default_log_retention/tf-lambda-iam-role.tf line 1, in resource "aws_iam_role" "log_retention":
resource "aws_iam_role" "log_retention" {
```

**Describe the solution you'd like**

"arn:aws:logs:*:*:*" should be flexible like "arn:${aws-resource-specific-region}:logs:*:*:*"
looking for other solutions as well.

**Comment by Clete2** [comment]
I think I can fix this by using `data "aws_partition"`. Standby

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition

**Comment by Clete2** [comment]
@gyaneshwar I think I have implemented what it takes to fix this in #11. There was a code change necessary on one of the Lambdas too.

I have no way to test in GovCloud or China. Would you mind testing v1.0.2 and reporting back any issues you encounter?

Thank you for opening and reporting this issue!

**Comment by gyaneshwar** [comment]
closing the ticket, it works well, but facing an issue with 1.0.2 reported in [here](https://github.com/StateFarmIns/terraform-aws-default-log-retention/issues/12).
